samples/java-security-usage/README.md

```diff
@@ -1,7 +1,83 @@
 # Description
-This sample is a Java back-end application running on the Cloud Foundry. On incoming requests it checks whether the user is authorized using the 
+This sample is a Java back-end application running on SAP Cloud Business Technology Platform. On incoming requests it checks whether the user is authorized using the 
 [`XsuaaTokenAuthenticator`](/java-security/src/main/java/com/sap/cloud/security/servlet/XsuaaTokenAuthenticator.java) which is defined in the [Java Security](../../java-security/) library.
 
+Follow the deployment steps for [Kyma/Kubernetes](#Deployment-on-Kyma/Kubernetes) or [Cloud Foundry](#Deployment-on-Cloud-Foundry).
+
+
+# Deployment on Kyma/Kubernetes
+<details>
+<summary>Expand this to follow the deployment steps</summary>
+
+- Compile the Java application
+- Build docker image and push to repository
+- Configure the deployment.yml
+- Deploy the application
+- Assign Role Collection to your user
+- Access the application
+
+## Compile the Java application
+Run maven to package the application
+```shell
+mvn clean package
+```
+## Build docker image and push to repository
+```shell script
+docker build -t <repositoryName>/<imageName> -f ./Dockerfile . 
+docker push <repositoryName>/<imageName>
+```
+
+## Configure the deployment.yml
+Replace the placeholders for image repository in deployment.yml with the one created in the previous step.
+
+## Deploy the application
+Deploy the application using [kubectl cli](https://kubernetes.io/docs/reference/kubectl/)
+```shell script
+kubectl apply -f ./k8s/deployment.yml -n <YOUR NAMESPACE>
+```
+
+## Cockpit administration tasks: Assign Role Collection to your user
+Finally, as part of your Identity Provider, e.g. SAP ID Service, assign the deployed Role Collection `JAVA_SECURITY_SAMPLE_Viewer` to your user as depicted in the screenshot below and as documented [here](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/9e1bf57130ef466e8017eab298b40e5e.html).
+
+![](../images/SAP_CP_Cockpit_AssignRoleCollectionToUser.png)
+
+Further up-to-date information you can get on sap.help.com:
+- [Maintain Role Collections](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/d5f1612d8230448bb6c02a7d9c8ac0d1.html)
+- [Maintain Roles for Applications](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/7596a0bdab4649ac8a6f6721dc72db19.html).
+
+## Access the application
+- Get an access token via `curl`. You can get the information to fill the placeholders from the service binding secret:
+```shell script
+kubectl get secret "xsuaa-service-binding" -o go-template='{{range $k,$v := .data}}{{"### "}}{{$k}}{{"\n"}}{{$v|base64decode}}{{"\n\n"}}{{end}}' -n <YOUR NAMESPACE>
+```
+
+```
+curl -X POST \
+  <<url>>/oauth/token \
+  -H 'Content-Type: application/x-www-form-urlencoded' \
+  -d 'client_id=<<clientid>>&client_secret=<<clientsecret>>&grant_type=password&username=<<your username>>&password=<<your password>>'
```

Comment:
does this fit to X.509 client credential setup?

Reply:
But wouldn't you fetch an access token with certificate and clientsecret isn't provided with secrets at all?

Reply:
no and i also noticed that this sample is not configured to use X.509 also in CF, should we then adapt both envs to use mTLS?

Reply:
that's right, in cf version its alos not yet adapted... we can also keep it here as it is for now...

Reply:
good finding right on time 💯